githubapp/client_creator.go

```diff
@@ -248,7 +254,11 @@ func (c *clientCreator) NewInstallationV4Client(installationID int64) (*githubv4
 
 func (c *clientCreator) NewTokenClient(token string) (*github.Client, error) {
 	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
-	tc := oauth2.NewClient(context.Background(), ts)
+	return c.NewTokenSourceClient(context.Background(), ts)
+}
+
+func (c *clientCreator) NewTokenSourceClient(ctx context.Context, ts oauth2.TokenSource) (*github.Client, error) {
+	tc := oauth2.NewClient(ctx, ts)
```

Comment:
This is where the context is passed.
https://cs.opensource.google/go/x/oauth2/+/a41e5a78:oauth2.go;l=342

Reply:
Looks like `oauth2.NewClient` passes the context to [`internal.ContextClient`](https://cs.opensource.google/go/x/oauth2/+/a41e5a78:internal/transport.go;l=23), which reads a value out of the context (if it exists) or returns `http.DefaultClient` otherwise. As far as I can see, the context is used for nothing else in `oauth2.NewClient` and once that function returns, there is nothing else that still holds a reference to the context.